gremlin-go/driver/client.go

```diff
@@ -42,38 +47,48 @@ type Client struct {
 	traversalSource string
 	logHandler      *logHandler
 	transporterType TransporterType
-	connection      *connection
+	connections     connectionPool
 	session         string
 }
 
 // NewClient creates a Client and configures it with the given parameters.
 func NewClient(url string, configurations ...func(settings *ClientSettings)) (*Client, error) {
 	settings := &ClientSettings{
-		TraversalSource: "g",
-		TransporterType: Gorilla,
-		LogVerbosity:    Info,
-		Logger:          &defaultLogger{},
-		Language:        language.English,
-		AuthInfo:        &AuthInfo{},
-		TlsConfig:       &tls.Config{},
-		Session:         "",
+		TraversalSource:              "g",
+		TransporterType:              Gorilla,
+		LogVerbosity:                 Info,
+		Logger:                       &defaultLogger{},
+		Language:                     language.English,
+		AuthInfo:                     &AuthInfo{},
+		TlsConfig:                    &tls.Config{},
+		NewConnectionThreshold:       4,
+		MaximumConcurrentConnections: runtime.NumCPU(),
+		Session:                      "",
 	}
 	for _, configuration := range configurations {
 		configuration(settings)
 	}
+
 	logHandler := newLogHandler(settings.Logger, settings.LogVerbosity, settings.Language)
-	conn, err := createConnection(url, settings.AuthInfo, settings.TlsConfig, logHandler)
+	var pool connectionPool
+	var err error
+	if settings.Session != "" {
+		pool, err = newLoadBalancingPool(url, settings.AuthInfo, settings.TlsConfig, settings.NewConnectionThreshold,
+			settings.MaximumConcurrentConnections, logHandler)
+	} else {
+		pool, err = newSingletonPool(url, settings.AuthInfo, settings.TlsConfig, logHandler)
+	}
```

Comment:
Could newLoadBalancingPool and newSingletonPool be combined into newPool where a singleton pool is just 1 & 1?

Reply:
Also is this backwards? if session is present, you are making a load balancing pool. We want a session to have a singleton pool, not load balancing.

Reply:
Was wondering about the same thing

Reply:
Basically I was wondering if we can change this so that is session != "" we update the connection threshold and maximum concurrent connections.

Then we just call one function to generate our pool